test/extensions/filters/http/wasm/wasm_filter_stress_test.cc

```diff
@@ -167,6 +167,114 @@ TEST_P(GrpcWasmStressTest, CalloutHappyPath) {
   EXPECT_EQ(counters["test_callout_successes"], requests_to_send);
 }
 
+TEST_P(GrpcWasmStressTest, CalloutErrorResponse) {
+  constexpr uint32_t connections_to_initiate = 30;
+  constexpr uint32_t requests_to_send = 30 * connections_to_initiate;
+  const std::string wasm_file = absl::StrCat(
+      TestEnvironment::runfilesDirectory(),
+      "/test/extensions/filters/http/wasm/test_data/grpc_callout_", wasmLang(), ".wasm");
+  // Must match cluster name in the wasm bundle:
+  const std::string callout_cluster_name{"callout_cluster"};
+
+  //
+  // Configure the wasm filter
+  //
+
+  config_helper_.addFilter(fmt::format(R"EOF(
+            name: envoy.wasm
+            config:
+              vm_config:
+                vm: "{}"
+                code:
+                  filename: "{}"
+                allow_precompiled: true
+)EOF",
+                                       wasmVM(), wasm_file));
+
+  //
+  // This test first sends a request from wasm to a callout cluster:
+  //    Client -> Envoy -> Wasm -> Callout Cluster
+  // Envoy then unconditionally forwards the request to the origin cluster
+  //              Envoy -> Origin Cluster
+  //
+  // Both Callout and Origin clusters have a single server that immediately
+  // returns a 200 OK response.
```

Comment:
Please update the comment.

Reply:
Good catch.  This reflects a copy and paste bug.  So...   I could go two ways on these test cases.  We could have lots of similar boiler plate copied into each test case (I know this sounds instantly bad - but it would make the tests cases more understandable without having yet another layer to understand) or I could factor out the common code and make it a bit harder to follow these things...

What would you recommend?

Reply:
Comments fixed

Reply:
For tests I think unfactored works better if you believe that the tests will be stable.  Typically tests are read more frequently by random folks addressing a single failure and learning additional abstractions has negative value.